perf.groovy

```diff
@@ -0,0 +1,120 @@
+
+def static getBuildJobName(def configuration, def os) {
+    return configuration.toLowerCase() + '_' + os.toLowerCase()
+}
+
+// Setup perflab tests runs
+[true, false].each { isPR ->
+    ['Windows_NT'].each { os ->
+        ['x64', 'x86'].each { arch ->
+            def architecture = arch
+            def jobName = "perf_perflab_${os}_${arch}"
+            def testEnv = ""
+            def newJob = job(Utilities.getFullJobName(project, jobName, isPR)) {
+
+                // Set the label.
+                label('windows_server_2016_clr_perf')
+                wrappers {
+                    credentialsBinding {
+                        string('BV_UPLOAD_SAS_TOKEN', 'CoreCLR Perf BenchView Sas')
+                    }
+                }
+
+                if (isPR) {
+                    parameters {
+                        stringParam('BenchviewCommitName', '\${ghprbPullTitle}', 'The name that you will be used to build the full title of a run in Benchview.  The final name will be of the form SDK <private|rolling> BenchviewCommitName')
+                    }
+                }
+
+                def configuration = 'Release'
+                def runType = isPR ? 'private' : 'rolling'
+                def benchViewName = isPR ? 'SDK private %BenchviewCommitName%' : 'SDK rolling %GIT_BRANCH_WITHOUT_ORIGIN% %GIT_COMMIT%'
+                def uploadString = '-uploadToBenchview'
+                def perfWorkingDirectory = '%WORKSPACE%\\artifacts\\${configuration}\\TestResults\\Performance'
+
+                steps {
+                    // Batch
+                    batchFile("powershell -NoProfile wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -OutFile \"${perfWorkingDirectory}\\nuget.exe\"")
+                    batchFile("if exist \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\" rmdir /s /q \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\"")
+                    batchFile("\"${perfWorkingDirectory}\\nuget.exe\" install Microsoft.BenchView.JSONFormat -Source http://benchviewtestfeed.azurewebsites.net/nuget -OutputDirectory \"${perfWorkingDirectory}\" -Prerelease -ExcludeVersion")
+                    // Do this here to remove the origin but at the front of the branch name as this is a problem for BenchView
+                    //    we have to do it all as one statement because cmd is called each time and we lose the set environment variable
+                    batchFile("if \"%GIT_BRANCH:~0,7%\" == \"origin/\" (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH:origin/=%\") else (set \"GIT_BRANCH_WITHOUT_ORIGIN=%GIT_BRANCH%\")\n" +
+                    "set \"BENCHVIEWNAME=${benchViewName}\"\n" +
+                    "set \"BENCHVIEWNAME=%BENCHVIEWNAME:\"=\"\"%\"\n" +
+                    "py \"${perfWorkingDirectory}\\Microsoft.BenchView.JSONFormat\\tools\\submission-metadata.py\" --name \"%BENCHVIEWNAME%\" --user-email \"dotnet-bot@microsoft.com\" --output \"${perfWorkingDirectory}\"\n" +
```

Comment:
I am fairly certain that when you use the output argument you need to also specify the file name, not just the out directory.

Reply:
Done...